olive/evaluator/olive_evaluator.py

```diff
@@ -128,6 +129,10 @@ def evaluate(
     ) -> MetricResult:
         metrics_res = {}
         for metric in metrics:
+            # use model io_config if user does not specify input_names and input_shapes
+            # only do this if data_config or dataloader is not provided
+            # priority: dataloader_func > data_config > user_config.input_names/input_shapes > model io_config
+            metric = OliveEvaluator.fill_metric_user_config_with_model_io(metric, model)
```

Comment:
I think it would be better and safer if we do not change the original object. We reuse the evaluator and metric objects for multiple models so I don't think the evaluation of one model should change the metric object. 

In general, I think we should not change any of our shared objects or object created from config jsons after config validation. It affects hashing+caching as well as sharing the object.

Reply:
Updated.